src/pages/workspace/accounting/qbo/export/QuickbooksCompanyCardExpenseAccountSelectCardPage.tsx

```diff
@@ -68,11 +69,18 @@ function QuickbooksCompanyCardExpenseAccountSelectCardPage({policy}: WithPolicyC
         (row: CardListItem) => {
             if (row.value !== exportCompanyCard) {
                 Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD, row.value);
-                Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD_ACCOUNT);
+                if (row.value === CONST.QUICKBOOKS_EXPORT_COMPANY_CARD.VENDOR_BILL) {
+                    Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_COMPANY_CARD_ACCOUNT);
+                    Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.EXPORT_ACCOUNT_PAYABLE, accountPayable?.[0]?.name);
+                    Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.QBO, CONST.QUICK_BOOKS_CONFIG.AUTO_CREATE_VENDOR);
```

Comment:
New record! 4 API calls 😅 cc @hayata-suenaga same question ^

Reply:
@s77rt is correct. This breaks the 1:1:1 API design and shouldn't be allowed.

Reply:
also, can you remind me why we added this field to the config object?
<img width="400" alt="Screenshot 2024-04-30 at 12 33 55 PM" src="https://github.com/Expensify/App/assets/98560306/76ffbb11-ceab-4d94-8483-7a776d28c1be">

Reply:
also this one
<img width="400" alt="Screenshot 2024-04-30 at 12 36 15 PM" src="https://github.com/Expensify/App/assets/98560306/61c4ce29-487b-48f5-8728-5ec79ddac0a2">

Reply:
were these values there before your PR? 🤔

Reply:
The changes in the backend will be done in [this GH issue](https://github.com/Expensify/Expensify/issues/390731) to update several fields of the connections configuration object, but the issue description might need to be updated. cc: @aldo-expensify

There are fields I don't recognized, so I'd wait for @narefyev91's confirmation

Reply:
yes these values were added in [the PR](https://github.com/Expensify/App/pull/40902/files).

<img width="400" alt="Screenshot 2024-04-30 at 12 41 59 PM" src="https://github.com/Expensify/App/assets/98560306/8bcdc771-6895-4b4b-a69d-71d4a7e3efd9">

Reply:
I think these fields don't exist on the backend 🤔

Reply:
We shouldn't be adding config fields that don't already exist. This is a sample working QBO config:

```
{
    "realmId": "2222",
    "credentials": {},
    "companyName": "Expensify",
    "autoSync": {
        "jobID": "11111111111",
        "enabled": true
    },
    "syncPeople": true,
    "syncItems": true,
    "markChecksToBePrinted": false,
    "reimbursableExpensesExportDestination": "bill",
    "nonReimbursableExpensesExportDestination": "credit_card",
    "reimbursableExpensesAccount": {
        "glCode": "",
        "name": "Accounts Payable",
        "currency": "USD",
        "id": "54"
    },
    "nonReimbursableExpensesAccount": {
        "glCode": "",
        "name": "Expensify Card Liability Account",
        "currency": "USD",
        "id": "135"
    },
    "autoCreateVendor": true,
    "hasChosenAutoSyncOption": true,
    "syncClasses": "REPORT_FIELD",
    "syncCustomers": "REPORT_FIELD",
    "syncLocations": "REPORT_FIELD",
    "exportDate": "LAST_EXPENSE",
    "lastConfigurationTime": 1714507864961,
    "syncTax": false,
    "enableNewCategories": true,
    "export": {
        "exporter": "aldo+testqbo56242@expensifail.com"
    },
    "nonReimbursableBillDefaultVendor": "NONE",
    "receivableAccount": {
        "glCode": "",
        "name": "Accounts Receivable",
        "currency": "USD",
        "id": "53"
    },
    "reimbursementAccountID": "93",
    "collectionAccountID": "93"
}
```